src/components/constants.ts

```diff
@@ -57,6 +58,28 @@ export const ACTION_CATEGORIES: ActionCategories = {
   date: [
     { name: 'todate', label: 'Convert text to date' },
     { name: 'fromdate', label: 'Convert date to text' },
+    { name: 'dateextract', label: 'Extract year from date', defaults: { operation: 'year' } },
```

Comment:
In terms of labels, I would rather shorten them and say "Extract year", "Extract month" etc.
We are in the Date widget so no need to repeat date everywhere